ls1intum/Artemis
# `Development`: Add a shared build job queue for local continuous integration

### Checklist
#### General


#### Server

### Motivation and Context


The Local CI build job queue is not distributed. For build jobs to be distributed accross server instances and being able to display the current build job queue in the client, we need a queue that is shared accross the server instances.

### Description

This PR implements a Hazelcast IQueue that can be seen and accessed by Hazelcast cluster members. Whenever a build job is triggered, a build job item including the participation id and if existing, the commit hash, is added to the queue. The service actively listens for new items in the queue. If a new item is added, the service checks whether the server instance has the capacity to process the build job. If so, the build job is processed. 
I adjusted some test cases to account for changes.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- Local VC CI Test Server (3 or 6)

1. As Instructor, create a programming exercise
2. Check, whether build jobs are processed correctly by confirming the build results
3. As Student, make a submission
4. Confirm build result

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
